pkg/kando/location.go

```diff
@@ -38,17 +47,86 @@ func newLocationCommand() *cobra.Command {
-	_ = cmd.MarkFlagRequired(profileFlagName)
+	cmd.PersistentFlags().StringP(repositoryServerFlagName, "r", "", "Pass a Repository Server CR as a JSON string (required for kopia based blueprints)")
 	return cmd
 }
 
 func pathFlag(cmd *cobra.Command) string {
 	return cmd.Flag(pathFlagName).Value.String()
 }
 
+// validateCommandArgs makes sure that we are getting exactly
+// one of --profile or --repository-server flags
+func validateCommandArgs(cmd *cobra.Command) error {
+	profileFlag := cmd.Flags().Lookup(profileFlagName).Value.String()
+	repositoryServerFlag := cmd.Flags().Lookup(repositoryServerFlagName).Value.String()
+	if profileFlag != "" && repositoryServerFlag != "" {
+		return errors.New("Either --profile or --repository-server can be provided")
+	}
+	if profileFlag == "" && repositoryServerFlag == "" {
+		return errors.New("Please provide either --profile or --repository-server as per the datamover you want to use")
+	}
+	return nil
+}
+
+func dataMoverFromCMD(cmd *cobra.Command, flag string) (datamover.DataMover, error) {
+	kopiaSnapshot, outputName := getKopiaSnapshotAndOutputNameFlag(cmd, flag)
+	dataMoverType, err := dataMoverTypeFromCMD(cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	switch dataMoverType {
+	case DataMoverTypeProfile:
+		profileRef, err := unmarshalProfileFlag(cmd)
+		if err != nil {
+			return nil, err
+		}
+		return datamover.NewProfileDataMover(profileRef, outputName, kopiaSnapshot), nil
+	case DataMoverTypeRepositoryServer:
+		repositoryServerRef, err := unmarshalRepositoryServerFlag(cmd)
+		if err != nil {
+			return nil, err
+		}
+		return datamover.NewRepositoryServerDataMover(repositoryServerRef, outputName, kopiaSnapshot), nil
+	default:
+		return nil, errors.New("Please provide either --profile or --repository-server as per the datamover you want to use")
+	}
+}
+
 func unmarshalProfileFlag(cmd *cobra.Command) (*param.Profile, error) {
 	profileJSON := cmd.Flag(profileFlagName).Value.String()
 	p := &param.Profile{}
 	err := json.Unmarshal([]byte(profileJSON), p)
 	return p, errors.Wrap(err, "failed to unmarshal profile")
 }
+
+func unmarshalRepositoryServerFlag(cmd *cobra.Command) (*param.RepositoryServer, error) {
+	repositoryServerJSON := cmd.Flag(repositoryServerFlagName).Value.String()
+	rs := &param.RepositoryServer{}
+	err := json.Unmarshal([]byte(repositoryServerJSON), rs)
+	return rs, errors.Wrap(err, "failed to unmarshal kopia repository server CR")
+}
+
+func dataMoverTypeFromCMD(c *cobra.Command) (DataMoverType, error) {
+	profile := c.Flags().Lookup(profileFlagName).Value.String()
+	if profile != "" {
+		return DataMoverTypeProfile, nil
+	}
+	repositoryServer := c.Flags().Lookup(repositoryServerFlagName).Value.String()
+	if repositoryServer != "" {
+		return DataMoverTypeRepositoryServer, nil
+	}
+	return "", errors.New("Please provide either --profile or --repository-server as per the datamover you want to use")
```

Comment:
@r4rajat we don't need to return error in this function. We are already validating the args in `validateCommandArgs()`

Reply:
Done !